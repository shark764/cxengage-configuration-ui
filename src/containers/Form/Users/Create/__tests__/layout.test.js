/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import UsersForm from '../layout';

describe('<UsersForm />', () => {
  it('renders create user', () => {
    const rendered = shallow(
      <UsersForm
        email="mockEmail"
        platformRoleId="mockPlatformRoleId"
        roleId="mockRoleId"
        firstName="mockFirstName"
        lastName="mockLastName"
        externalId="1234"
        workStationId="1234"
        active={false}
        platformRoles={[]}
        tenantRoles={[]}
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders create user saving and inherited', () => {
    const rendered = shallow(
      <UsersForm
        email="mockEmail"
        platformRoleId="mockPlatformRoleId"
        roleId="mockRoleId"
        firstName="mockFirstName"
        lastName="mockLastName"
        externalId="1234"
        workStationId="1234"
        active={true}
        platformRoles={[]}
        tenantRoles={[]}
        isSaving={true}
        inherited={true}
        userHasUpdatePermission={true}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders create user without create permision', () => {
    const rendered = shallow(
      <UsersForm
        id="57083781-332d-11e6-8dd4-c88eee4d9f61"
        email="mockEmail"
        platformRoleId="mockPlatformRoleId"
        roleId="mockRoleId"
        firstName="mockFirstName"
        lastName="mockLastName"
        externalId="1234"
        workStationId="1234"
        active={true}
        platformRoles={[]}
        tenantRoles={[]}
        isSaving={false}
        inherited={false}
        userHasUpdatePermission={false}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
