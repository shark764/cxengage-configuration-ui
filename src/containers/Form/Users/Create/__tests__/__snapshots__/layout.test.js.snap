// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersForm /> renders create user 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.div>
    <Connect(DetailWrapper)
      open={true}
    >
      <Styled(DetailHeader)
        text="Tenant Details"
      />
      <InputField
        componentType="input"
        disabled={false}
        id="frm-users-work-station-id"
        inputType="text"
        label="Workstation ID"
        name="workStationId"
      />
      <SelectField
        disabled={false}
        id="frm-users-role-id"
        label="Tenant Role *"
        name="roleId"
        options={Array []}
        required={true}
      />
      <ToggleField
        disabled={false}
        id="frm-users-invite-now"
        label="Invite Now"
        name="inviteNow"
      />
    </Connect(DetailWrapper)>
    <Connect(DetailWrapper)
      open={true}
    >
      <Styled(DetailHeader)
        text="Login Details"
      />
      <InputField
        componentType="input"
        disabled={false}
        id="frm-users-email"
        inputType="text"
        label="Email *"
        name="email"
      />
      <SelectField
        disabled={false}
        id="frm-users-platform-role-id"
        label="Platform Role *"
        name="platformRoleId"
        options={Array []}
        required={true}
      />
      <SelectField
        disabled={false}
        id="frm-users-no-password-id"
        label="Platform Authentication"
        name="noPassword"
        options={
          Array [
            Object {
              "label": "Use Tenant Default: Enabled",
              "value": "null",
            },
            Object {
              "label": "Enabled",
              "value": true,
            },
            Object {
              "label": "Disabled",
              "value": false,
            },
          ]
        }
        required={true}
      />
      <SelectField
        disabled={false}
        id="frm-users-default-identity-provider-id"
        label="Single Sign On Identitiy Provider"
        name="defaultIdentityProvider"
        required={true}
      />
    </Connect(DetailWrapper)>
  </styled.div>
</form>
`;

exports[`<UsersForm /> renders create user saving and inherited 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.div>
    <Connect(DetailWrapper)
      open={true}
    >
      <Styled(DetailHeader)
        text="Tenant Details"
      />
      <InputField
        componentType="input"
        disabled={true}
        id="frm-users-work-station-id"
        inputType="text"
        label="Workstation ID"
        name="workStationId"
      />
      <SelectField
        disabled={true}
        id="frm-users-role-id"
        label="Tenant Role *"
        name="roleId"
        options={Array []}
        required={true}
      />
      <ToggleField
        disabled={true}
        id="frm-users-invite-now"
        label="Invite Now"
        name="inviteNow"
      />
    </Connect(DetailWrapper)>
    <Connect(DetailWrapper)
      open={true}
    >
      <Styled(DetailHeader)
        text="Login Details"
      />
      <InputField
        componentType="input"
        disabled={false}
        id="frm-users-email"
        inputType="text"
        label="Email *"
        name="email"
      />
      <SelectField
        disabled={true}
        id="frm-users-platform-role-id"
        label="Platform Role *"
        name="platformRoleId"
        options={Array []}
        required={true}
      />
      <SelectField
        disabled={true}
        id="frm-users-no-password-id"
        label="Platform Authentication"
        name="noPassword"
        options={
          Array [
            Object {
              "label": "Use Tenant Default: Enabled",
              "value": "null",
            },
            Object {
              "label": "Enabled",
              "value": true,
            },
            Object {
              "label": "Disabled",
              "value": false,
            },
          ]
        }
        required={true}
      />
      <SelectField
        disabled={true}
        id="frm-users-default-identity-provider-id"
        label="Single Sign On Identitiy Provider"
        name="defaultIdentityProvider"
        required={true}
      />
    </Connect(DetailWrapper)>
  </styled.div>
</form>
`;

exports[`<UsersForm /> renders create user without create permision 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.div>
    <Connect(DetailWrapper)
      open={true}
    >
      <Styled(DetailHeader)
        text="Tenant Details"
      />
      <InputField
        componentType="input"
        disabled={true}
        id="frm-users-work-station-id"
        inputType="text"
        label="Workstation ID"
        name="workStationId"
      />
      <SelectField
        disabled={true}
        id="frm-users-role-id"
        label="Tenant Role *"
        name="roleId"
        options={Array []}
        required={true}
      />
      <ToggleField
        disabled={true}
        id="frm-users-invite-now"
        label="Invite Now"
        name="inviteNow"
      />
    </Connect(DetailWrapper)>
    <Connect(DetailWrapper)
      open={true}
    >
      <Styled(DetailHeader)
        text="Login Details"
      />
      <InputField
        componentType="input"
        disabled={false}
        id="frm-users-email"
        inputType="text"
        label="Email *"
        name="email"
      />
      <SelectField
        disabled={true}
        id="frm-users-platform-role-id"
        label="Platform Role *"
        name="platformRoleId"
        options={Array []}
        required={true}
      />
      <SelectField
        disabled={true}
        id="frm-users-no-password-id"
        label="Platform Authentication"
        name="noPassword"
        options={
          Array [
            Object {
              "label": "Use Tenant Default: Enabled",
              "value": "null",
            },
            Object {
              "label": "Enabled",
              "value": true,
            },
            Object {
              "label": "Disabled",
              "value": false,
            },
          ]
        }
        required={true}
      />
      <SelectField
        disabled={true}
        id="frm-users-default-identity-provider-id"
        label="Single Sign On Identitiy Provider"
        name="defaultIdentityProvider"
        required={true}
      />
    </Connect(DetailWrapper)>
  </styled.div>
</form>
`;
